satellite/repair/repairer/segments.go

```diff
@@ -78,7 +79,12 @@ func (repairer *SegmentRepairer) Repair(ctx context.Context, path storj.Path) (s
 	// Read the segment pointer from the metainfo
 	pointer, err := repairer.metainfo.Get(ctx, path)
 	if err != nil {
-		return storj.ErrObjectNotFound.Has(err), Error.Wrap(err)
+		if storage.ErrKeyNotFound.Has(err) {
```

Comment:
I wonder whether the correct error should be `ErrObjectNotFound` or `ErrSegmentNotFound` or something else... it feels like the service should be returning a better error.